x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -2142,6 +2212,130 @@ private static Expression typeSpecificConvert(ConvertFunction convert, Source so
         }
     }
 
+    private static class ResolveUnmapped extends ParameterizedAnalyzerRule<LogicalPlan, AnalyzerContext> {
+
+        @Override
+        protected LogicalPlan rule(LogicalPlan plan, AnalyzerContext context) {
+            return switch (context.unmappedResolution()) {
+                case UnmappedResolution.FAIL -> plan;
+                case UnmappedResolution.NULLIFY -> nullify(plan);
+                case UnmappedResolution.LOAD -> throw new IllegalArgumentException("unmapped fields resolution not yet supported");
+            };
+        }
+
+        private LogicalPlan nullify(LogicalPlan plan) {
+            List<UnresolvedAttribute> nullifiedUnresolved = new ArrayList<>();
+            List<UnresolvedAttribute> unresolved = new ArrayList<>();
+            Holder<Boolean> unresolvable = new Holder<>(false);
+            var resolved = plan.transformDown(p -> unresolvable.get() ? p : switch (p) {
```

Comment:
Why do we go down the whole plan? Because we're inside a ParameterizedAnalyzerRule, we will already have walked all plan children before reaching here.

Using ResolveRefs as a comparison, I'd expect something like:
- check if the plan has unresolved children; if so, skip (children has attributes that couldn't be nullified for whatever reason)
- check for new unresolved attributes in the current plan node, only
- if any are found, do whatever we need to do to nullify them.